www/__tests__/uploadService.test.ts

```diff
@@ -0,0 +1,46 @@
+//this is never used in production right now
+//however, tests are still important to make sure the code works
+//at some point we hope to restore this functionality
+
+import {uploadFile} from "../js/control/uploadService";
+import { mockLogger } from '../__mocks__/globalMocks';
+import { mockDevice, mockGetAppVersion, mockCordova, mockFile } from "../__mocks__/cordovaMocks";
+import { mockFileSystem } from "../__mocks__/fileSystemMocks";
+
+mockDevice();
+mockGetAppVersion();
+mockCordova();
+
+mockLogger();
+mockFile(); //mocks the base directory 
+mockFileSystem(); //comnplex mock, allows the readDBFile to work in testing
+
+//use this message to verify that the post went through
+let message = "";
+
+global.fetch = (url: string, options: {method: string, headers: {}, body: string}) => new Promise((rs, rj) => {
```

Comment:
can you clarify what this does? I understand that if there are options, you return "sent..." and otherwise, you return "mock data" after a timeout, but I don't understand why there is a timeout only in the `else` and not in the `if`

Reply:
I've added comments about what the difference between the if and the else are, for some more clarity. I also added a timeout to the if, you're right, it doesn't make sense to have it one place and not the other.